app/app.go

```diff
@@ -527,3 +528,69 @@ func (app *BandApp) AutoCliOpts() autocli.AppOptions {
 		ConsensusAddressCodec: authcodec.NewBech32Codec(sdk.GetConfig().GetBech32ConsensusAddrPrefix()),
 	}
 }
+
+func isInvalidCheckTxExecution(resp *abci.ResponseCheckTx, checkTxErr error) bool {
+	return resp == nil || resp.Code != 0 || checkTxErr != nil
+}
+
+// CheckTx returns a CheckTx handler that wraps a given CheckTx handler and evicts txs that are not
+// in the app-side mempool on ReCheckTx.
+func (app *BandApp) CheckTx(req *abci.RequestCheckTx) (*abci.ResponseCheckTx, error) {
+	// decode tx
+	tx, err := app.BaseApp.TxDecode(req.Tx)
+	if err != nil {
+		return sdkerrors.ResponseCheckTxWithEvents(
+			fmt.Errorf("failed to decode tx: %w", err),
+			0,
+			0,
+			nil,
+			false,
+		), nil
+	}
+
+	mempool := app.Mempool().(*mempool.Mempool)
+
+	isReCheck := req.Type == abci.CheckTxType_Recheck
```

Comment:
Recheck is a single word, use Recheck
```suggestion
	isRecheck := req.Type == abci.CheckTxType_Recheck
```